peachjam/models/citations.py

```diff
@@ -78,3 +84,36 @@ def for_target_works(cls, work):
             .filter(target_work=work)
             .order_by("citing_work__title")
         )
+
+
+class CitationProcessing(SingletonModel):
+    processing_date = models.DateField(_("processing date"), null=True, blank=True)
+
+    def update_processing_date(self, date):
+        self.processing_date = date
+        self.save()
+
+    def re_extract_citations(self):
+        """Extract citations for all documents dated on or after the processing date."""
+        from peachjam.models import CoreDocument
+        from peachjam.tasks import extract_citations
+
+        later_documents = CoreDocument.objects.filter(
```

Comment:
This could load 100,000 documents, including all their content. Use `only("pk")`. You may also want to use `.iterator()` on in the for loop.